csrc/multidevice/lower_communication.cpp

```diff
@@ -397,10 +391,19 @@ void lowerToReduceScatter(
   params.redOp = getC10dReduceOpType(op_type);
   params.team = mesh.vector();
   params.dst_bufs = {output_tensor};
+  // TODO: Clean up once allocation domain for sharded tensors is
+  // better supported.
+  auto red_axis = output_tv->getReductionAxis().value();
+  auto shard_axis = getShardedAxis(output_tv);
+  if (red_axis <= shard_axis) {
+    shard_axis++;
+  }
```

Comment:
I'm sure this and code around it are doing something cool, but I cannot immediately understand them! Can you add some code comment?